src/hotspot/share/services/mallocTracker.cpp

```diff
@@ -281,7 +260,7 @@ void* MallocTracker::record_malloc(void* malloc_base, size_t size, MEMFLAGS flag
   assert(((size_t)memblock & (sizeof(size_t) * 2 - 1)) == 0, "Alignment check");
 
 #ifdef ASSERT
-  if (level > NMT_minimal) {
+  if (level >= NMT_summary) {
```

Comment:
```suggestion
  if (level > NMT_off) {
```

This way, we don't care where `NMT_summary` is in the enum. We enter this code on all paths when NMT is enabled?

Reply:
Actually, maybe even call to `enabled()` here?